pkg/servicescm/servicescm.go

```diff
@@ -158,14 +167,23 @@ func Generate(name, namespace string, data *Data) (*core.ConfigMap, error) {
 	}
 	servicesConfigMap.Data[filesKey] = string(jsonFiles)
 
+	if len(data.EnvironmentVars) > 0 {
+		jsonEnvVars, err := json.Marshal(data.EnvironmentVars)
```

Comment:
Do we need to sanity check the `EnvironmentVars` data? Is it possible for the user try and sneak in a custom variable that WICD will end up setting on the host? I realize WMCO will revert the ConfigMap to the expected state but there could be a Window  where WICD ends up adding the environment variable. It is not as if we will end up removing the added variable in that case.

Reply:
Hm, the services CM is immutable so sneaking this in would be tough

Reply:
Yes I am aware it is immutable but the user could delete and create one with the same name but different content.

Reply:
Anyway, the scenario you described will never occur as WICD only sets variables that are in the SupportedProxyVars set. It does not iterate over all env vars present in the services CM

Reply:
That is incorrect now that you have stated it explicitly. WICD should be blindly following what is set in the ConfigMap. WMCO should control what goes into the ConfigMap. IOW, WMCO sets the "policy" and WICD implements it and that boundary should be maintained.